components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -459,6 +459,35 @@ public String getPublishedDefaultVersion(APIIdentifier apiid) throws APIManageme
 
 
     /**
+     * This method is used to save the wsdl file in the registry
+     * This is used when user starts api creation with a soap endpoint
+     *
+     * @param api api object
+     * @throws APIManagementException
+     * @throws RegistryException
+     */
+    private void updateWsdl(API api) throws APIManagementException, RegistryException {
+
+        registry.beginTransaction();
```

Comment:
Please note that the code is already fixed after a code review with the team. The commit id is 041980d7db64a5ee61428d9f6dc63aea37d14359 and you can refer to the updated code [1]
[1] - https://github.com/madusankapremaratne/carbon-apimgt/commit/041980d7db64a5ee61428d9f6dc63aea37d14359